ui/components/multichain/pages/send/components/domain-input-resolution-cell.tsx

```diff
@@ -0,0 +1,244 @@
+import React, {
+  ///: BEGIN:ONLY_INCLUDE_IF(build-flask)
+  useContext,
+  ///: END:ONLY_INCLUDE_IF
+  useRef,
+  useEffect,
+  useState,
+} from 'react';
+import PropTypes from 'prop-types';
+///: BEGIN:ONLY_INCLUDE_IF(build-flask)
+import { I18nContext } from '../../../../../contexts/i18n';
+///: END:ONLY_INCLUDE_IF
+import Confusable from '../../../../ui/confusable';
+import {
+  AvatarAccount,
+  ///: BEGIN:ONLY_INCLUDE_IF(build-flask)
+  AvatarIcon,
+  AvatarIconSize,
+  BadgeWrapper,
+  Box,
+  IconName,
+  ///: END:ONLY_INCLUDE_IF
+  Text,
+} from '../../../../component-library';
+import {
+  AlignItems,
+  ///: BEGIN:ONLY_INCLUDE_IF(build-flask)
+  BackgroundColor,
+  BorderColor,
+  Display,
+  IconColor,
+  ///: END:ONLY_INCLUDE_IF
+  TextColor,
+  TextVariant,
+} from '../../../../../helpers/constants/design-system';
+import { ellipsify } from '../../../../../pages/confirmations/send/send.utils';
+import Tooltip from '../../../../ui/tooltip';
+
+type DomainInputResolutionCellArgs = {
+  domainType: string;
+  address: string;
+  protocol?: string;
+  domainName: string;
+  resolvingSnap?: string;
+  onClick: () => void;
+};
+
+export const DomainInputResolutionCell = ({
+  domainType,
+  address,
+  domainName,
+  ///: BEGIN:ONLY_INCLUDE_IF(build-flask)
+  resolvingSnap = '',
+  ///: END:ONLY_INCLUDE_IF
+  onClick,
+  protocol,
+}: DomainInputResolutionCellArgs) => {
+  ///: BEGIN:ONLY_INCLUDE_IF(build-flask)
+  const t: (key: string, params: unknown[]) => string = useContext(I18nContext);
+  ///: END:ONLY_INCLUDE_IF
+  const titleRef = useRef<null | HTMLDivElement>(null);
+  const breakpointRef = useRef<null | number>(null);
+  const [isTitleOverflowing, setIsTitleOverflowing] = useState(false);
+
+  useEffect(() => {
+    if (!titleRef.current) {
+      return;
+    }
+
+    let isOverflowing =
+      titleRef.current.offsetWidth < titleRef.current.scrollWidth;
+    const breakpointLength = titleRef.current.textContent?.length;
+
+    if (isOverflowing && !breakpointRef.current && breakpointLength) {
+      breakpointRef.current = breakpointLength;
+    }
+
+    if (!isOverflowing) {
+      if (breakpointRef.current) {
+        if (domainName.length >= breakpointRef.current) {
+          isOverflowing = true;
+        } else {
+          isOverflowing = false;
+          breakpointRef.current = null;
+        }
+      }
+    }
+
+    if (isOverflowing !== isTitleOverflowing) {
+      setIsTitleOverflowing(isOverflowing);
```

Comment:
What is the goal of all of this?

Reply:
Trying to avoid setting state when it's not necessary

Reply:
Apologies, I should have clarified. What are we trying to achieve by detecting the overflow and using different titles?

Reply:
Oh, so that we can display the overflowing title in a tooltip. We want to display the whole resolved name because there are snaps that will fuzzy match, so what you're typing in is not necessarily what you're getting in the resolution.